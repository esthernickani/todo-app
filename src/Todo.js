import React, { useState } from "react";

const Todo = ({ id, todo, removeTodo }) => {
    console.log(id)
    const handleRemove = () => removeTodo(id)

    return (
        <span>
            <p>{todo}</p>
            <button onClick={handleRemove}>X</button>
        </span>
    )
}

export default Todo;