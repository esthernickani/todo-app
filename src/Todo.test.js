import React from "react";
import Todo from "./Todo";
import { render } from "@testing-library/react";
import { v4 as uuid } from 'uuid';

it("renders without crashing", function() {
    const id = uuid()
    const mockRemove = jest.fn()
    render(<Todo todo="Wash clothes" id={id} key={id} removeTodo={mockRemove} />)
})

it("matches snapshot", function() {
    const id = uuid()
    const mockRemove = jest.fn()

    const { asFragment } = render(<Todo todo="Wash clothes" id={id} key={id} removeTodo={mockRemove} />)
    expect(asFragment()).toMatchSnapshot()
})