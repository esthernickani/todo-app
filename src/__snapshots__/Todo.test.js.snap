// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <span>
    <p>
      Wash clothes
    </p>
    <button>
      X
    </button>
  </span>
</DocumentFragment>
`;
